### DISCUSSION TITLE
Title auto-generation does not respect keep_alive setting

### DISCUSSION THREAD

**Discussion by amirhossein-ka** [open-webui/open-webui#4418] [General]
# Bug Report

## Description

**Bug Summary:**
When using keep alive setting to change time that models is loaded on gpu (in my case -1), on the new chats on the first prompt the model reloads and its keep alive time resets to 5 min, leading to performance issues.

**Steps to Reproduce:**

1. Change keep alive setting to some other value than 5m.
2. Start a new chat and give a prompt to model.
3. Monitor the model keep alive time using `watch -n 0.1 ollama ps` or `nvtop` to check gpu memory usage.

**Expected Behavior:**
Title auto-generation should respect keep-alive setting and do not cause to models get reloaded.

**Actual Behavior:**
This feature ignores keep alive setting.

## Environment

- **Open WebUI Version:** v0.3.10
- **Ollama (if applicable):** 0.3.0

- **Operating System:** Ubuntu 24.04

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Screenshots (if applicable):**
![image](https://github.com/user-attachments/assets/1fe32a91-12ef-43c3-865c-5d7a54e921a3)

## Installation Method

open-webui installed using docker and ollama is installed directly on host machine.

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

**Comment by justinh-rahb** [comment]
Set the `OLLAMA_KEEP_ALIVE` environment variable for Ollama itself. The WebUI keepalive parameter is known to be unreliable, envvar _always_ works.

**Reply by amirhossein-ka** [reply]
I set this environment in ollama service file and restarted it, then removed keep alive setting from open-webui panel but the problem still persists, it is same as image above.

**Reply by justinh-rahb** [reply]
What does `ollama ps` output in terminal after a model is loaded by WebUI?

**Reply by amirhossein-ka** [reply]
It says that model is loaded until *Forever*, but when i'm monitoring its output after creating new chat and send some prompt, the model reloads (just output will be empty for a second and then it shows up again), and for the second prompt same thing happens. but after that it says that model will be there until *Forever*. or until a new chat is opened.


Edit:
The other thing that i noticed is that the loaded model for title generation uses less memory than normal one.

**Reply by justinh-rahb** [reply]
Modifying the `num_ctx` parameter (and some others, but this is the usual culprit) will cause models to reload, and memory usage to change.

**Reply by amirhossein-ka** [reply]
Thank you that was the culprit for sure 😄😄. 
By the way, is it possible to change `num_ctx` without causing problems like this ?

It still feels like the title generator does not follow all settings defined in open-webui.

**Reply by justinh-rahb** [reply]
Seems this issue is still open:
- #3106
